### DISCUSSION TITLE
NvStrapsReBar EFI Variable Error: Access is denied

### DISCUSSION THREAD

**Discussion by githubnowx21** [terminatorul/NvStrapsReBar#26] [Q&A]
I am on an X99 Motherboard (Rampage V Extreme), I successfully injected the .ffs file to the bios image, flashing the new bios was successful, I also used that Uefi patch tool to patch the bios, I tried flashing that bios but wouldn't go through, however, when using the NvstrapReBar tool, while upon saving the changes, I get an error:

Error saving configuration to NvStrapsReBar EFI variable: Access is denied.

Using a Titan V, thanks.

**Comment by xCuri0** [comment]
Uefipatch fixes this

**Reply by githubnowx21** [reply]
Patching the bios image with uefi patch works, its after the image is patched, its creates another file with a .patched extension, I tried flashing the motherboard with this new image, I tried both with and without the .patched extension, says something about the bios image is not a valid file.

EDIT:

I found that, on Asus mobo's, I have to use the Flashback procedure, so doing the normal bios flashing through EZ flash won't work with modified bios?

Question, which ReBar .ffs do I use? ReBarDxe.ffs or NvStrapsReBar.ffs?

Also, the instructions doesn't say where to add the NvStrapReBar.ffs, do I inject it in the same location as the ReBarDxe.ffs? where it says PciBus?

Also, just heads up, I tried using the Flashback while flashing the bios, couldn't get it to work, even with stock factory bios.

I tried the following methods: Fat16 (Also did convert mbr within diskpart & convert part size to =4000) & Fat 32(Did both convert mbr as well as convert part size=4000), this was done on a Sandisk Cruzer 16gb, 2.0 usb stick.

Do I have to use DSDT Patching?

EDIT:

Ok, I got the bios flashed successfully, I had to rename the file to R5E.cap, however, despite of this and disabling CSM within the bios, I still get that access denied error within NvStrapsReBar tool, this is with UEFIpatch (After injecting the ReBarDxe.ffs within the bios image).

I also used ReBarState.exe, I get this error when entering a variable:

Failed to write ReBarState UEFI variable.

Tried both: NvStrapsReBar.ffs & ReBarDxe.ffs.

So now what do I do now?

**Comment by SebVig** [comment]
I posted a detailed guide on [Win-Raid forum](https://winraid.level1techs.com/t/offer-asus-rampage-v-extreme-modded-bioses/31405/508) how I made my BIOS for this board, maybe it can help you to add both NvStrapsReBar.ffs & ReBarDxe.ffs modules.
I can do it for you if needed.

**Reply by githubnowx21** [reply]
Someone named Powerfull helped me, I got right up to the end where I was going to flash the bios through flashback, it couldn't flash at all, did 3 blinks then quit, did this with both Fat16 & Fat32 (Convert mbr and used: Create part primary size=4000).

I used a specific modified bios for the same exact motherboard I am using that was up on Win-Raid site, this one actually worked, however, it says ReBar is disabled, even though it is working.

Here is a link to the steps that powerfull wanted me to do, again, everything was successful, just the flashing part was not successful, wouldn't even start.

https://winraid.level1techs.com/t/offer-asus-rampage-v-extreme-modded-bioses/31405/508

**Reply by githubnowx21** [reply]
Yes please, I went through alot to get this to work, all unsuccessful, I even tried that bios mod for the exact motherboard that I own (Asus Rampage V Extreme), although the bios flashed properly, soon I get into windows, both ReBarState.exe and that NvStrapReBar.exe, works, however, under GPU-z, ReBar still shows disabled and under Display properties > Resources, there is no Large Memory present, so its not working properly.

Could you please make one working for me? Motherboard I am using: Asus Rampage V Extreme / Bios version 4101 (Latest beta bios),

The one Xcuri0 recommends me to use is the NvStrapsReBar.ffs, instead of using ReBarDxe.ffs, thanks.

**Reply by SebVig** [reply]
I will read the NvStrapsReBar page, then I will try to build the BIOS with NvStrapsReBar.ffs.

**Reply by githubnowx21** [reply]
It could very well be that Volta based gpu's are not supported, natively the Titan V, ReBar ain't supported right off the back, I figured after doing some Frame Generation mods, why not some ReBar action, and yes, Frame Generations mods works great with the Titan V, some games no, lots of tinkering around, sorry for the bit off topic, thanks for the help everyone.
